pkg/chartutil/chartfile.go

```diff
@@ -58,3 +62,37 @@ func SaveChartfile(filename string, cf *chart.Metadata) error {
 	}
 	return ioutil.WriteFile(filename, out, 0644)
 }
+
+// IsChartDir validate a chart directory.
+//
+// Checks for a valid Chart.yaml.
+func IsChartDir(dirName string) (bool, error) {
+	if fi, err := os.Stat(dirName); err != nil {
+		return false, err
+	} else if !fi.IsDir() {
+		return false, fmt.Errorf("%q is not a directory", dirName)
+	}
+
+	chartYaml := filepath.Join(dirName, "Chart.yaml")
```

Comment:
It seems you could get by with just the one stat of the complete dirName/Chart.yaml. Im not sure loading it and throwing it away is necessary since an invalid YAML will be caught later.